### PR TITLE
task: add strict rules to enforce pythonic naming

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: catparser/parserutils.py [hunk 1]
```diff
@@ -3,8 +3,9 @@
 from .CatsParseException import CatsParseException
 
 REGEXES = {
-    'user_type_name': re.compile(r'^[A-Z][a-zA-Z0-9]*$'),
-    'property_name': re.compile(r'^[a-z_][a-zA-Z0-9_]*$'),
+    'user_type_name': re.compile(r'^[A-Z]+[a-z0-9][a-zA-Z0-9]*$'),  # UpperCamelCase
```
Comment (Jaguar0625) [inline catparser/parserutils.py:6]: not strictly UpperCamelCase in order to not conflict with TITLE_CASE

File: catparser/parserutils.py [hunk 2]
```diff
@@ -20,27 +21,42 @@ def _match_regex_or_throw(regex_key, line):
     return match
 
 
-def require_user_type_name(type_name):
-    """Raises an exception if the specified name is not a valid user type name"""
-    _match_regex_or_throw('user_type_name', type_name)
-    return type_name
+class TypeNameChecker:
```
Comment (Jaguar0625) [inline catparser/parserutils.py:24]: added to improve imports

File: tests/constants.py [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
-
 INT_TYPE_TUPLES = [('int8', 1, 'signed'), ('int16', 2, 'signed'), ('int32', 4, 'signed'), ('int64', 8, 'signed')]
 UINT_TYPE_TUPLES = [('uint8', 1, 'unsigned'), ('uint16', 2, 'unsigned'), ('uint32', 4, 'unsigned'), ('uint64', 8, 'unsigned')]
 PRIMITIVE_TYPE_TUPLES = INT_TYPE_TUPLES + UINT_TYPE_TUPLES
 BUILTIN_TYPE_TUPLES = PRIMITIVE_TYPE_TUPLES + [
     ('binary_fixed(32)', 32, 'unsigned'), ('binary_fixed(0x20)', 32, 'unsigned'), ('binary_fixed(25)', 25, 'unsigned')
 ]
+
+
+class NameConstants:
```
Comment (Jaguar0625) [inline tests/constants.py:9]: added to improve imports

File: tests/test_CatsParser.py [hunk 4]
```diff
@@ -659,19 +659,20 @@ def test_cannot_parse_struct_with_unsupported_type_condition(self):
 
     def _assert_can_parse_struct_with_const_member(self, disposition):
         # Act:
+        const_property_name = 'tuple_size' if 'reserved' == disposition else 'TUPLE_SIZE'
```
Comment (Jaguar0625) [inline tests/test_CatsParser.py:662]: due to differences in const vs reserved naming, needed this construction in a bunch of places, not sure if it should be refactored somehow?

File: tests/test_StructParser.py [hunk 5]
```diff
@@ -279,48 +281,65 @@ def _assert_parse_exception(self, line):
...
-                    {'name': 'red', 'type': 'ColorShade', 'value': value, 'disposition': disposition})
+                    '{} = make_{}(ColorShade, {})'.format(name, disposition, value),
+                    {'name': name, 'type': 'ColorShade', 'value': value, 'disposition': disposition})
 
     def test_cannot_parse_binary_fixed_type_constant(self):
         for disposition in self.SUPPORTED_DISPOSITIONS:
+            name = 'foo' if 'reserved' == disposition else 'FOO'
             SingleLineParserTestUtils(StructConstParserFactory, self).assert_parse_exception(
-                'foo = make_{}(binary_fixed(25), 123)'.format(disposition))
+                '{} = make_{}(binary_fixed(25), 123)'.format(name, disposition))
```
Comment (gimre-xymcity) [inline tests/test_StructParser.py:318]: those tests became quite hard to read 😬

**Review by gimre-xymcity — APPROVED** [review]

File: tests/test_StructParser.py (hunk 5 above)
Comment (Jaguar0625) [reply tests/test_StructParser.py:318]: 🤷‍♂️

**Pull request merged**